lib/git_worktree.rb

```diff
@@ -258,6 +258,22 @@ def checkout
     end
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  #
+  def check_connection(url)
+    require 'net/ping/http'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
+    url = url.sub(':', '/').sub('git@', 'https://www.') if url.start_with?('git@')
+
+    # Secure endpoints only as port 80 may not be enabled for outgoing traffic.
+    url = url.sub('http', 'https') if url.start_with?('http://')
+
+    $log.debug("pinging url '#{url}' to verify network connection")
+
+    Net::Ping::HTTP.new(url).ping?
```

Comment:
will this raise an exception or just return a bool?  The reason I'm asking is because I'd expect a method named check_connection to raise on a bad connection.  If not, then maybe this method should be named `check_connection?` instead?  Not sure...what do you think?.

Reply:
Returns a bool.

Reply:
Ok, follow up comment then, here: https://github.com/ManageIQ/manageiq/pull/20759#issuecomment-719619805